src/components/document-metadata-modal.js

```diff
@@ -56,47 +67,65 @@ function DocumentMetadataModal({
+    cloningStrategy,
+    cloningTargetRoomId
+  }), [mode, initialDocumentMetadata, documentToClone, cloningStrategy, cloningTargetRoomId]);
 
-  const initialQualityMetadataValues = mode === DOCUMENT_METADATA_MODAL_MODE.update
-    ? {
-      review: initialDocumentMetadata.review,
-      verified: initialDocumentMetadata.verified,
-      allowedOpenContribution: initialDocumentMetadata.allowedOpenContribution
-    }
-    : {
-      review: '',
-      verified: false,
-      allowedOpenContribution: DOCUMENT_ALLOWED_OPEN_CONTRIBUTION.metadataAndContent
-    };
-
-  const initialValues = {
-    title: initialDocumentMetadata.title || t('newDocument'),
-    description: initialDocumentMetadata.description || '',
-    slug: initialDocumentMetadata.slug || '',
-    tags: initialDocumentMetadata.tags || [],
-    language: initialDocumentMetadata.language || getDefaultLanguageFromUiLanguage(uiLanguage),
-    generateSequence: false,
-    sequenceCount: 2,
-    useTemplateDocument: false,
-    cloningStrategy: CLONING_STRATEGY.cloneWithinArea,
-    cloningTargetRoomId: '',
-    ...initialQualityMetadataValues
+  const publicContextPermissions = {
+    canArchive: hasUserPermission(user, permissions.MANAGE_ARCHIVED_DOCS),
+    canVerify: hasUserPermission(user, permissions.VERIFY_DOC),
+    canReview: hasUserPermission(user, permissions.REVIEW_DOC),
+    canRestrictOpenContribution: hasUserPermission(user, permissions.RESTRICT_OPEN_CONTRIBUTION)
   };
+  const hasPublicContextPermissions = Object.values(publicContextPermissions).some(value => value);
 
   const cloningOptions = getCloningOptions({ mode, documentToClone, availableRooms, t });
-
   const allowedOpenContributionOptions = getAllowedOpenContributionOptions({ t });
 
-  const validationRules = getValidationRules({ t });
+  const defaultTemplateDocumentId = settings.templateDocument?.documentId || null;
+  const canUseTemplateDocument = mode === DOCUMENT_METADATA_MODAL_MODE.create && !!defaultTemplateDocumentId;
+  const canCreateSequence = mode === DOCUMENT_METADATA_MODAL_MODE.create && allowMultiple;
+  const canSelectCloningStrategy = mode === DOCUMENT_METADATA_MODAL_MODE.clone && cloningOptions.strategyOptions.length > 1;
+
+  const validationState = useMemo(
+    () => getValidationState({ t, title, description, slug, tags, cloningStrategy, cloningTargetRoomId }),
+    [t, title, description, slug, tags, cloningStrategy, cloningTargetRoomId]
+  );
+
+  const resetStates = useCallback(() => {
+    setTitle(initialDocumentMetadata.title || t('newDocument'));
+    setDescription(initialDocumentMetadata.description || '');
+    setSlug(initialDocumentMetadata.slug || '');
+    setTags(initialDocumentMetadata.tags || []);
+    setTagOptions(composeTagOptions(initialDocumentMetadata.tags));
+    setLanguage(initialDocumentMetadata.language || getDefaultLanguageFromUiLanguage(uiLanguage));
+    setPublicContext(initialDocumentMetadata.publicContext || getDefaultPublicContext());
```

Comment:
It would feel cleaner if we'd `deepClone`d this object (although technically irrelevant). Also, please see my comment at the end of this PR on the cloning case.

Reply:
Cleaner maybe no, but definitely safer! Cloned it, thanks.